File: android/src/main/kotlin/com/usercentrics/sdk/flutter/bridge/ShowSecondLayerBridge.kt
package com.usercentrics.sdk.flutter.bridge

import com.usercentrics.sdk.flutter.api.*
import com.usercentrics.sdk.flutter.serializer.deserializeBannerSettings
import com.usercentrics.sdk.flutter.serializer.serialize

internal class ShowSecondLayerBridge(
    private val assetsProvider: FlutterAssetsProvider,
    private val activityProvider: FlutterActivityProvider,
    private val banner: UsercentricsBannerProxy = UsercentricsBannerProxySingleton,
) : MethodBridge {

    override val name: String
        get() = "showSecondLayer"

    override fun invoke(call: FlutterMethodCall, result: FlutterResult) {
        assert(name == call.method)

        val argsMap = call.arguments as Map<*, *>

        banner.showSecondLayer(
            context = activityProvider.provide(),
            bannerSettings = argsMap["bannerSettings"].deserializeBannerSettings(
                assetsProvider,
                activityProvider
            ),
            showCloseButton = argsMap["showCloseButton"] as Boolean,
        ) { response ->
            result.success(response?.serialize())
        }
    }
}

File: ios/usercentrics_sdk/Sources/usercentrics_sdk/Bridge/ShowSecondLayerBridge.swift
import UsercentricsUI
import Flutter

struct ShowSecondLayerBridge : MethodBridge {

    let assetProvider: FlutterAssetProvider
    let bannerProxy: UsercentricsBannerProxyProtocol
    let name: String = "showSecondLayer"

    init(assetProvider: FlutterAssetProvider,
         bannerProxy: UsercentricsBannerProxyProtocol = UsercentricsBannerProxy()) {
        self.assetProvider = assetProvider
        self.bannerProxy = bannerProxy
    }

    func invoke(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
        assert(call.method == name)
        let argsDict = call.arguments as! Dictionary<String, Any>
        let bannerSettings = BannerSettings(from: argsDict["bannerSettings"], assetProvider: assetProvider)
        bannerProxy.showSecondLayer(bannerSettings: bannerSettings) { response in
            result(response.serialize())
        }
    }
}
